src/line_coordinates.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+#ifndef CATA_SRC_LINE_COORDINATES_H
+#define CATA_SRC_LINE_COORDINATES_H
+// Cannot include this in line.h since that would require a mutual header dependency C can't handle.
+// The purpose of this header is to provide operations on the derived point types. Note that the
+// implementations are still located in line.cpp.
+
+#include <vector>
+
+#include "coordinates.h"
+
+// t and t2 decide which Bresenham line is used.
+std::vector<tripoint_bub_ms> line_to( const tripoint_bub_ms &loc1, const tripoint_bub_ms &loc2,
+                                      int t = 0, int t2 = 0 );
```

Comment:
Isn't this already implemented by these? The latter two parameters should be added there though.

https://github.com/CleverRaven/Cataclysm-DDA/blob/81ede3cc7667289d9c6e7c98dd80c98c6b5d6d03/src/coordinates.h#L796-L809

Reply:
Yes, it looks like this SHOULD be the solution. However, when adding the parameters with default values the macro stupidly causes users to blow up due to "ambiguity" whether the 2D or 3D version of line_to in line.h is called, even though the 2D version can't be called with the 3D parameters.
The problem here is that the 2D version has a single optional parameter while the 3D version has 2 optional parameters, but the macro still think it can call the 2D one with two additional parameters (which probably is incorrect if there are any 2D callers).

Adding a similar macro (or "template") specifying Tripoint still causes ambiguity.
I'm still fumbling around, but it wouldn't hurt if some expert advice could be had.

Apart from that, I think it's a bad idea to have this stuff in coordinates.h. This header ought to just contain the coordinate related stuff, not include extra stuff from line.h causing it not to be possible to be included in a lot of headers due to circular inclusion. However, I'm not going to try to sort that out without a clear prod.